### DISCUSSION TITLE
Exposing the attribute data-command

### DISCUSSION THREAD

**Discussion by georgekaran** [ariakit/ariakit#1119] [Questions] [answered]
When upgrading the Ariakit to v2, we notice that a new attribute was added to the DOM which is the `data-command`.

Taking a more in-depth look we realized that this attribute is an implementation detail to avoid duplicated elements.
The problem is that in our application some snapshots are now failing due to this new attribute. 

Should we expose this attribute to the end-use since it's an implementation detail? Can we omit this property when it's empty?

**Comment by diegohaz** [comment, marked as answer]
> The problem is that in our application some snapshots are now failing due to this new attribute.

I don't understand how this is a problem. If you're doing snapshot tests on components, you're already testing implementation details. They will fail whenever we (or you, in your application) add or remove an attribute from the component, which may be a totally valid and non-breaking change.

Just update the snapshot.

> Should we expose this attribute to the end-use since it's an implementation detail? Can we omit this property when it's empty?

The `data-command` prop is always empty. It's there just to tell the component already implements [`Command`](https://ariakit.org/examples/command) so we don't fire events twice.

All the props passed to the underlying element, including `role` and event listeners, should be considered implementation details. You shouldn't rely on them in your tests or they will eventually break.

The exception is a few data attributes that will be stated in the docs, such as `data-active-item`, `data-focus-visible` etc.
